app/repositories/OptOutSessionDataRepository.scala

```diff
@@ -0,0 +1,118 @@
+import utils.OptOutJourney
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OptOutSessionDataRepository @Inject()(val repository: UIJourneySessionDataRepository) {
+
+  def saveIntent(intent: TaxYear)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    OptionT(repository.get(hc.sessionId.get.value, OptOutJourney.Name)).
+      map(journeySd => journeySd.copy(optOutSessionData = journeySd.optOutSessionData.map(_.copy(selectedOptOutYear = Some(intent.toString))))).
+      flatMap(journeySd => OptionT.liftF(repository.set(journeySd))).
+      getOrElse(false)
+  }
+
+  def recallOptOutProposition()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[OptOutProposition]] = {
+    repository.get(hc.sessionId.get.value, OptOutJourney.Name) map { sessionData =>
+      for {
+        data <- sessionData
+        optOutData <- data.optOutSessionData
+        contextData <- optOutData.optOutContextData
+        currentYear <- TaxYear.getTaxYearModel(contextData.currentYear)
+      }
+      yield createOptOutProposition(
+        currentYear,
+        contextData.crystallisationStatus,
+        stringToStatus(contextData.previousYearITSAStatus),
+        stringToStatus(contextData.currentYearITSAStatus),
+        stringToStatus(contextData.nextYearITSAStatus))
+    }
+  }
+
+  def fetchSavedIntent()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[TaxYear]] = {
+    repository.get(hc.sessionId.get.value, OptOutJourney.Name) map { sessionData =>
+      for {
+        data <- sessionData
+        optOutData <- data.optOutSessionData
+        selected <- optOutData.selectedOptOutYear
+        parsed <- TaxYear.getTaxYearModel(selected)
+      } yield parsed
+    }
+  }
+
+  def initialiseOptOutJourney(oop :OptOutProposition)(implicit hc: HeaderCarrier): Future[Boolean] = {
+    val data = UIJourneySessionData(
+      sessionId = hc.sessionId.get.value,
+      journeyType = OptOutJourney.Name,
+      optOutSessionData = Some(OptOutSessionData(Some(buildOptOutContextData(oop)), selectedOptOutYear = None))
+    )
+    repository.set(data)
+  }
+
+  private def buildOptOutContextData(oop: OptOutProposition): OptOutContextData = {
+    OptOutContextData(
+      oop.currentTaxYear.taxYear.toString,
+      oop.previousTaxYear.crystallised,
+      statusToString(oop.previousTaxYear.status),
+      statusToString(oop.currentTaxYear.status),
+      statusToString(oop.nextTaxYear.status))
+  }
+
+}
+
+case class OptOutContextData(currentYear: String,
+                             crystallisationStatus: Boolean,
+                             previousYearITSAStatus: String,
+                             currentYearITSAStatus: String,
+                             nextYearITSAStatus: String)
+
+object OptOutContextData {
+  implicit val format: OFormat[OptOutContextData] = Json.format[OptOutContextData]
+
+  def statusToString(status: ITSAStatus): String =
```

Comment:
I think 
- should pull changes from main 
- and move 'statusToString', 'stringToStatus' to the new OptOutRepositoryService object 
- and reuse these for both OptOut and OptIn
- and delete repositories.ITSAStatusRepositorySupport.scala from main (which I don't like being there)

Reply:
Good spot! I missed that one when merging main. I've pushed up a change that uses the centralised functions in ITSAStatusRepositorySupport. I think this is the right thing to do as ITSAStatusRepositorySupport is then shared between both Opt Out and Opt In.